addons/mail/static/src/core/common/message_reaction_list.js

```diff
@@ -0,0 +1,58 @@
+/* @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+
+export class MessageReactionList extends Component {
+    static template = "mail.MessageReaction.List";
+    static props = {
+        id: { type: Number, required: true },
+        close: { type: Function, required: true },
+        reaction: { type: Object, required: true },
+        openReactionMenu: { type: Function, required: true },
+    };
+
+    getReactionSummary() {
+        const reaction = this.props.reaction;
+        const [firstUserName, secondUserName, thirdUserName] = reaction.personas.map(
+            ({ name, displayName }) => name || displayName
+        );
+        switch (reaction.count) {
+            case 1:
+                return _t("%s has reacted with %s", firstUserName, reaction.content);
+            case 2:
+                return _t(
+                    "%s and %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    reaction.content
+                );
+            case 3:
+                return _t(
+                    "%s, %s, %s have reacted with %s",
+                    firstUserName,
+                    secondUserName,
+                    thirdUserName,
+                    reaction.content
+                );
+            default:
+                return {
+                    names: _t("%s, %s, %s and ", firstUserName, secondUserName, thirdUserName),
+                    highlightText: _t("%s other", reaction.personas.length - 3),
+                    content: _t(" person have reacted with %s", reaction.content),
```

Comment:
Do not break a sentence like that, other languages do not necessarily have the same split as in English.
Instead, you should enumerate the different formats explicitly, each having their own `_t("")` of full sentence.

Note that the 4 and 4+ are different, due to singular/plural of "other(s)"